Sources/Apodini/Semantic Model Builder/EndpointParameter.swift

```diff
@@ -30,14 +30,26 @@ struct EndpointParameter {
             }
     }
     
+    static func create(from pathComponent: _PathComponent, label: String) -> EndpointParameter? {
+        guard let info: TypeInfo = try? typeInfo(of: type(of: pathComponent)) else {
+            return nil
+        }
+        return create(from: info, source: pathComponent, label: label)
+    }
+    
     static func create(from requestInjectable: RequestInjectable, label: String) -> EndpointParameter? {
         guard let info: TypeInfo = try? typeInfo(of: type(of: requestInjectable)) else {
             return nil
         }
+        return create(from: info, source: requestInjectable, label: label)
+    }
+    
+    /// TODO: Why does this not work with `create<T>(from source: T, ...) { typeInfo(of: type(of: source)) ... }`
```

Comment:
@PSchmiedmayer @Supereg Do you have a better idea for this? It does not seem to work with a straightforward generic function, as the `TypeInfo` will then always be the original `protocol` (e.g., `RequestInjectable` or `_PathComponent`) rather than `Parameter<.>`.

Reply:
> It does not seem to work with a straightforward generic function, as the TypeInfo will then always be the original protocol (e.g., RequestInjectable or _PathComponent) rather than Parameter<.>.

Such Type information for generics is always determined at **compile time** (at the location of the call). As create is called few lines above for `RequestInjectable` or `_PathComponent` those types are used.

In #53 I incorporated the visitor pattern for RequestInjectables to preserve the that type information. I could remove the dependency on the Runtime framework here.

Reply:
Perfect, I'm looking at it right now.